wal.go

```diff
@@ -401,14 +401,15 @@ func (w *SegmentWAL) Truncate(mint int64, keep func(uint64) bool) error {
 	csf.Sync()
 	csf.Close()
 
+	candidates[0].Close() // need close before remove on platform windows
```

Comment:
Is not closing really causing issues on windows or are you just assuming that it is more strict about requiring things to be closed explicitly? I'd like to avoid platform/OS specific comments unless there's really a tricky caveat.

Reply:
yes there are issues!
every change is to make sure, it pass test case.

Reply:
you can comment some change, and run test on windows. 
it will fail.